celest/units/quantity.py

```diff
@@ -0,0 +1,297 @@
+            return self._data == other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if isinstance(other, Quantity):
+            if self._unit.dimension != other._unit.dimension:
+                raise ArithmeticError(
+                    "Unit dimensions must match for comparison.")
+            return self._data < other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __le__(self, other):
+        return self.__eq__(other) or self.__lt__(other)
+
+    def __gt__(self, other):
+        if isinstance(other, Quantity):
+            if self._unit.dimension != other._unit.dimension:
+                raise ArithmeticError(
+                    "Unit dimensions must match for comparison.")
+            return self._data > other.to(self._unit)
+        else:
+            return NotImplemented
+
+    def __ge__(self, other):
+        return self.__eq__(other) or self.__gt__(other)
+
+    @property
+    def data(self) -> Any:
+        """
+        Return the data of the quantity.
+        """
+        return self._data
+
+    @property
+    def unit(self) -> Union[Unit, CompoundUnit]:
+        """
+        Return the unit of the quantity.
+        """
+        return self._unit
+
+    @property
+    def dimension(self) -> str:
+        """
+        Return the dimension string of the quantity.
+        """
+        return self._unit.dimension
+
+    def to(self, new_unit: Union[Unit, CompoundUnit]) -> Any:
+        """Return the Quantity data in the new unit.
+
+        Parameters
+        ----------
+        new_unit : Unit, CompoundUnit
+            New unit with the same dimension as the current unit.
+
+        Returns
+        -------
+        Any
+            The Quantity data in the new unit.
+
+        Examples
+        --------
+        >>> quantity_in_meters = Quantity(5, u.m)
+        >>> quantity_in_meters.to(u.km)
+        0.005
+        """
+
+        return self._convert_to(new_unit).data
+
+    def _convert_to(self, new_unit: Union[Unit, CompoundUnit]) -> "Quantity":
+        """Return a new Quantity object with the new unit.
+
+        Parameters
+        ----------
+        new_unit : Unit, CompoundUnit
+            New unit with the same dimension as the current unit.
+
+        Returns
+        -------
+        Quantity
+            A new Quantity object with the new unit.
+
+        Examples
+        --------
+        Initialize the `Quantity` object:
+
+        >>> quantity_in_meters = Quantity(5, u.m)
+
+        Convert to kilometers:
+
+        >>> quantity_in_kilometers = quantity_in_meters._convert_to(u.km)
+        """
+
+        if self._unit.dimension != new_unit.dimension:
+            raise ValueError(
+                "New unit has a different dimension then the current unit.")
+
+        def _get_unit_scale(unit):
+            if isinstance(unit, CompoundUnit):
```

Comment:
Consider changing the `CompoundUnit.scale` property to return the scale of the compound unit rather than unity.